integration/tests/cook/test_basic.py

```diff
@@ -1242,12 +1242,23 @@ def instance_query():
             resp = util.retry_jobs(self.cook_url, retries=4, failed_only=True, jobs=jobs)
             self.assertEqual(201, resp.status_code, resp.text)
             jobs = util.query_jobs(self.cook_url, True, uuid=jobs).json()
-            # We expect both jobs to be running now.
-            # The first job (which we killed and retried) should have 3 retries remaining
-            # (the attempt before resetting the total retries count is still included).
-            job_details = f"Job details: {json.dumps(jobs[0], sort_keys=True)}"
+            # We expect both jobs to be waiting or running now.
+            job_details = f"Job details: {json.dumps(jobs[0], sort_keys=True, indent=2)}"
+            self.logger.info(job_details)
             self.assertIn(jobs[0]['status'], ['waiting', 'running'], job_details)
-            self.assertEqual(3, jobs[0]['retries_remaining'], job_details)
+            non_mea_culpa_failure_exists = any(not i['reason_mea_culpa'] for i in jobs[0]['instances'])
```

Comment:
nitpick: I would just add some more utility functions to get a failed non-mea culpa job and not have the branch

Reply:
Done

Reply:
Done